### ISSUE TITLE
[Bug]: Adding and removing memberships to a user does not work correctly

### ISSUE THREAD

**Issue by hifabienne** [zitadel/zitadel#4859]
As an admin I want to be able to add and remove user memberships on instance, organisation, project and project grant level.

**Acceptance Criteria**
- [x] When I change a membership and it works successfully the entry is updated in the list
- [x] When I change a membership and an error happens, the error is shown and the entry is not updated in the list
- [x] When I click the "x" on the last role of a membership the membership will be removed instead of changed

### Describe the bug

A customer reported that when he likes to remove a role, by clicking on the x of the role in the role list (user detail memberships) he gets an error "invalid UpdateProjectGrantMemberRequest.ProjectId: value length must be between 1 and 200 runes, inclusive"
This only happens on that one user. 

As I tried to reproduce I could add a role the first time, when i wanted to remove one by clicking the "x" all despite one role were removed. When I tried to add again some role, it didn't woke "Errors.Org.MemberInvalid (Org-4N8es)"

### To reproduce

1. add some memeberships to a user
2. go to the detail of the user
3. go to the memberships
4. click the "x" on one role

### Version

2.14.2

**Comment by hifabienne** [comment]
I have investigated a little bit and saw a few problems in the frontend.
1. If I change some memberships and get an error, the list entry is still updated. So It shows me what I changed, but in fact didnt change in the backend
2. I am able to click the "x" on the last role of a membership, but this results in an error since it is not allowed to have an empty membership. We should delete the membership if it is the last role
